lib/tasks_private/openstack.rake

```diff
@@ -34,4 +34,65 @@ namespace :vcr do
     Rake::Task['vcr:spec:run'].invoke
     Rake::Task['vcr:credentials:obfuscate'].invoke
   end
+
+  task :rerecord_v2 => [] do
+    require 'fog/openstack'
+
+    host = YAML.load_file("config/secrets.yml").dig("test", "openstack", "hostname")
+    port = YAML.load_file("config/secrets.yml").dig("test", "openstack", "port")
+    username = YAML.load_file("config/secrets.yml").dig("test", "openstack", "userid")
+    password = YAML.load_file("config/secrets.yml").dig("test", "openstack", "password")
+
+    `rm spec/vcr_cassettes/manageiq/providers/openstack/cloud_manager/refresher.yml`
+    `rm spec/vcr_cassettes/manageiq/providers/openstack/network_manager/refresher.yml`
+
+    compute = Fog::Compute.new(:provider => "OpenStack", :openstack_auth_url => "https://#{host}:#{port}/v3/", :openstack_username => username, :openstack_api_key => password, :openstack_project_name => "admin", :openstack_domain_id => 
+      "default", :connection_options => {:ssl_verify_peer => false})
+
+    network = Fog::Network.new(:provider => "OpenStack", :openstack_auth_url => "https://#{host}:#{port}/v3/", :openstack_username => username, :openstack_api_key => password, :openstack_project_name => "admin", :openstack_domain_id => 
+        "default", :connection_options => {:ssl_verify_peer => false})
+
+    puts "Creating resources..."
+    cn = network.create_network(:name => "test-network")
+    cn_id = cn.data.dig(:body, "network", "id")
+    tenant_id = cn.data.dig(:body, "network", "tenant_id")
+    cs = network.subnets.create(:network_id => cn_id, :cidr => "10.0.0.0/21", :ip_version => 4)
+    router = network.create_router("test-router")
+    router_id = router.data.dig(:body, "router", "id")
+    sg = network.create_security_group(:name => "test-sg-group", :description => "test description", :tenant_id => tenant_id)
+    sg_id = sg.data.dig(:body, "security_group", "id")
+
+    image = compute.images.first
+    flavor = compute.flavors.get 1
+    vm = compute.servers.create(:name => 'test-server', :flavor_ref => flavor.id, :image_ref => image.id, :nics => [{:net_id => cn_id}])
+    key = compute.create_key_pair("test-key")
+    vol = compute.create_volume("test-vol", "test description", 1)
+    vol_id = vol.data.dig(:body, "volume", "id")
+    agg = compute.create_aggregate("test-aggregate")
+    agg_id = agg.data.dig(:body, "aggregate", "id")
+
+    puts "Generating VCRs..."
+    `bundle exec rspec spec/models/manageiq/providers/openstack/cloud_manager/refresher_spec.rb`
+    `bundle exec rspec spec/models/manageiq/providers/openstack/network_manager/refresher_spec.rb`
+
+    puts "Cleaning up resources..."
+    compute.delete_server(vm.id)
+    compute.delete_key_pair("test-key")
+    compute.delete_volume(vol_id)
+    compute.delete_aggregate(agg_id)
+
+    10.times do
+      sleep(10)
+    end
+
+    network.delete_subnet(cs.id)
+
+    10.times do
+      sleep(10)
+    end
+
+    network.delete_network(cn_id)
```

Comment:
Added delays to allow the completion of deleting resources that are dependent on each other (vms, subnets, and cloud networks)